ee/modules/650-runtime-audit-engine/hack/far-converter/main.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2023 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gosimple/slug"
+	"github.com/iancoleman/strcase"
+	"gopkg.in/yaml.v3"
+)
+
+// RawRule represents decoded rules in Falco format.
+type RawRule map[string]interface{}
+
+// FalcoAuditRule is a structure to encode FalcoAuditRules custom resources.
+type FalcoAuditRule struct {
+	ApiVersion string
+	Kind       string
+	Metadata   Metadata
+	Spec       FalcoAuditRuleSpec
+}
+
+type FalcoAuditRuleSpecRule struct {
+	Rule  Rule  `yaml:"rule,omitempty"`
+	Macro Macro `yaml:"macro,omitempty"`
+	List  List  `yaml:"list,omitempty"`
+}
+
+type FalcoAuditRuleSpec struct {
+	RequiredEngineVersion         int                      `yaml:"requiredEngineVersion,omitempty"`
+	RequiredK8sAuditPluginVersion string                   `yaml:"requiredK8sAuditPluginVersion,omitempty"`
+	Rules                         []FalcoAuditRuleSpecRule `yaml:"rules"`
+}
+
+type Metadata struct {
+	Name string
+}
+
+type Rule struct {
+	Name      string        `yaml:"name"`
+	Condition string        `yaml:"condition"`
+	Desc      string        `yaml:"desc"`
+	Output    string        `yaml:"output"`
+	Priority  string        `yaml:"priority"`
+	Enabled   bool          `yaml:"enabled,omitempty"`
+	Tags      []interface{} `yaml:"tags,omitempty"`
+	Source    string        `yaml:"source,omitempty"`
+}
+
+type Macro struct {
+	Name      string `yaml:"name"`
+	Condition string `yaml:"condition"`
+}
+
+type List struct {
+	Name  string        `yaml:"name"`
+	Items []interface{} `yaml:"items"`
+}
+
+func main() {
+	var input string
+	flag.StringVar(&input, "input", "", "Input file with Falco rules.")
+
+	flag.Parse()
+
+	var rules []RawRule
+
+	log.Printf("Convert rules from %q", input)
+
+	data, err := os.ReadFile(input)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = yaml.Unmarshal(data, &rules)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	res, err := yaml.Marshal(convert(input, rules))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println(string(res))
+}
+
+func convert(path string, rules []RawRule) FalcoAuditRule {
+	result := FalcoAuditRule{
+		ApiVersion: "deckhouse.io/v1alpha1",
+		Kind:       "FalcoAuditRules",
+		Metadata: Metadata{
+			Name: nameFromPath(path),
+		},
+		Spec: FalcoAuditRuleSpec{},
+	}
+
+	for _, r := range rules {
+		if v, ok := r["required_engine_version"]; ok {
+			result.Spec.RequiredEngineVersion = v.(int)
+			continue
+		}
+
+		if v, ok := r["required_plugin_versions"]; ok {
+			for _, p := range v.([]interface{}) {
+				plugin := p.(map[string]interface{})
+				if plugin["name"].(string) == "k8saudit" {
+					result.Spec.RequiredK8sAuditPluginVersion = plugin["version"].(string)
```

Comment:
NIT: Rule to CRD is made with golang
but CRD to Rule is made with python

Meanwhile, there are too many type casting here. Don't you want to make it also with python?

Reply:
The only reason I avoided python (my initial idea was also to stick to python everywhere in this module) is that managing dependencies for python is complicated. Users must download the right interpreter version and use a venv for environment isolation. At some point, I started thinking that maybe using python in a docker container is a good idea but ended with the go version of the script.

Do you think it is worth rewriting it in python? If so, I can play around `docker run` and some python script.

Reply:
I added TODO to the list

Reply:
Didn't get the idea. Can you elaborate?